plane_spotter/scripts/notify.py

```diff
@@ -43,21 +62,61 @@ class Config:
     notification_backend: Any = MISSING
 
 
-defaults = [
-    {"notification_backend": "twitter"},
-    {"adsb_backend": "adsbexchange"},
-]
+defaults = []
 
 cs = ConfigStore.instance()
 cs.store(group="adsb_backend", name="adsbexchange_schema", node=ADSBExchangeConfig)
-cs.store(group="notification_backend", name="twitter_schema", node=Twitter)
+cs.store(group="notification_backend", name="twitter_api_schema", node=TwitterAPI)
+cs.store(
+    group="notification_backend", name="twitter_selenium_schema", node=TwitterSelenium
+)
 cs.store(name="config_schema", node=Config)
 cs.store(name="airplane_schema", node=Airplane)
 
 
 logger = get_logger(__name__)
 
 
+def _main_loop(
+    adsb_backend: ADSBExchange,
+    geolocator: Geolocator,
+    twitter_selenium: _TwitterSelenium,
+    cfg: Config,
+    loop_interval: int = 120,
+    log=logger,
+):
+    last_known_ident = None
+    while True:
+        response = adsb_backend.aircraft_last_position_by_hex_id(
+            hex_id=cfg.airplane["icao_hex_id"]
+        ).json()
+        lat = response["lat"]
+        lon = response["lon"]
+        log.info(f"Plane last known location", lat=lat, lon=lon)
+
+        nearest_airport = geolocator.lookup_airport(
+            coordinates=(lat, lon), max_distance=cfg.search_radius
+        )
+        if nearest_airport is None:
+            log.info("not near any known airport")
+            return
+
+        log.info("Nearest airport info:")
+        log.info("\n" + json.dumps(nearest_airport, indent=4) + "\n")
+
+        ident = nearest_airport["ident"]
+        name = nearest_airport["name"]
+        if ident != last_known_ident:
+            log.info("airplane spotted near new airport")
+            now = datetime.datetime.now(datetime.timezone.utc)
+            twitter_selenium.send(message=f"Airplane spotted near {name} at {now}")
+            last_known_ident = ident
```

Comment:
Apparently I forgot to commit my latest changes that include airport info in the tweet